### ISSUE TITLE
[NAME OF FORM]: Issue description

### ISSUE THREAD

**Issue by HasanSifat** [bcgov/digital-journeys#573]
**User Story**
As an ADM after selecting the employee name, I want to see the personal details section of the form for that employee to be pre-populated so that I can confirm that I have chosen the right employee for nomination. 

**Tasks**

- [ ] Hook API endpoints to the personal details section 

- [ ] All fields in the section get pre-populated with info from the selected employee name field. 

**Acceptance Criteria**
this will be considered complete when the personal details section of the form gets pre-populated with info according to the selection made in the employee name field.

**Comment by bhumin-fw** [comment]
@HasanSifat 
I'm adding a field map from employee API and form data. Please correct me if it does not match.

**Form field: API field**
Employee Number: EMPLID
Employee Position Title: position_title
Employee Work City: input.office_city
Division: level2_division
Branch: level3_branch

For your reference, 
We do receive below fields from the API. (I did mask few data with xxx)
Its key value may help you to correct any mapping on the above table.
{
            "BUSINESS_UNIT": "BCxx",
            "ClassificationGroup": "Enforcement & Corrections",
            "DEPTID": "0xx-1xxx",
            "EFFDT": "2022-01-31",
            "EFFSEQ": 0,
            "EMPLID": "xxxxx5",
            "EMPL_CLASS": "B",
            "EMPL_CTG": "R",
            "EMPL_RCD": 0,
            "EMPL_STATUS": "A",
            "GUID": "xxxxxxxxxx3C3B51F63xxxxxxxxxx",
            "HIRE_DT": "1993-04-26",
            "IDIR": "xxxxxxxx",
            "JOBCODE": "xxxx00",
            "JOB_FUNCTION": "xx2",
            "JobCodeDescGroup": "Correctional Service",
            "OccupationalGroup": "Specialist, Professional",
            "Organization": "Justice",
            "PAYGROUP": "STD",
            "address1": "xxxxx maclure rd",
            "address2": " ",
            "appointment_status": "Regular",
            "city": "Abbotsford",
            "country": "CAN",
            "date_deleted": null,
            "date_updated": "2022-08-08T07:02:48.000-07:00",
            "email": "xxxk.xxxxxko@gov.bc.ca",
            "employee_status_long_description": "Active",
            "first_name": "Mark",
            "job_function_employee_group": "xxxEU",
            "jobcode_desc": "Correction Serv 18 ADLT CUST",
            "last_name": "xxxxxko",
            "level1_program": "BC Corrections",
            "level2_division": "Corr-Adult Custody",
            "level3_branch": "ADULT CUST-FMCC SUM",
            "level4": "Ford Mountain Corr Centre",
            "middle_name": "L",
            "name": "xxxx ko xxxx",
            "office_address1": "57657 Chilliwack Lake Rd.",
            "office_address2": " ",
            "office_city": "Sardis",
            "office_country": "CAN",
            "office_location_code": "xxxxP102",
            "office_phone": "",
            "office_postal": "V2R 2P1",
            "office_stateprovince": "BC",
            "phone": "xxx xxx 2533",
            "position_number": "xxx12xxx",
            "position_title": "Correction Serv 18 ADLT CUST",
            "postal": "V2T 5S6",
            "stateprovince": "BC",
            "supervisor_email": "xxxson.xxxxxren@gov.bc.ca",
            "supervisor_emplid": "xxxx27",
            "supervisor_name": "xxxxson xxxxren",
            "supervisor_position_number": "0006xxxx",
            "supervisor_position_title": "Assist Deputy Warden,Staffing",
            "supervisor_first_name": "xxxxxx",
            "supervisor_last_name": "xxxxxxx"
        }

**Comment by Stella-Archer** [comment]
we have the green light to hook up the end points

**Comment by bhumin-fw** [comment]
@Stella-Archer @HasanSifat 
It is ready to test in the DEV environment.
Ticket 572 can be tested with this one. Employee search API required authentication user with role `sl-review-adm` on a keylock.

**Comment by HasanSifat** [comment]
I've done one round of tests. The work city field is somehow being populated with the wrong info.

**Comment by HasanSifat** [comment]
I've looked at the data fields and have the following concerns- 
1. ministry fields need to be prepopulated 
2. ADM/supervisor name needs to be pre-populated
3. ADM/Supervisor email field needs to be pre-populated 

![image.png](https://images.zenhubusercontent.com/61e5b78d9a5a1eb2ba7a92c3/5f6450fe-b328-4b87-8c69-896ab3b3c968)

**Comment by HasanSifat** [comment]
@bhumin-fw if you look at my screenshot above my peoplesoft info for work location is saved as victoria. But somehow it's showing up as vancouver.

**Comment by bhumin-fw** [comment]
@HasanSifat We have discussed that we will pre-populate the ADM name and email from the logged-in user that's why it's blank. Let me update the API to populate it from the ODS data and pre-fill it for the selected employee.

Regarding the data that we are receiving from API. here is the full object. for name **Hasan Sifat**
In this object city is Victoria but the office_city is Vancouver.
Please do let me know if we should select **city** value instead of **office_city** value to populate **Employee Work City** filed in SL review form 2.0
{
            "BUSINESS_UNIT": "BC100",
            "ClassificationGroup": "Senior Admin & Research",
            "DEPTID": "100-35xx",
            "EMPLID": "1747xx",
            **"city": "Victoria",**
            "email": "Hasxx.xxxat@gov.bc.ca",
            "first_name": "Hasan",
            "last_name": "Sifat",
            "level2_division": "Operational Strategies & Integration Services ",
            "level3_branch": "Op Start & Integration Serv",
            "name": "Hasan Sifat",
            "office_address1": "802-865 HORNBY STREET",
            "office_address2": " ",
            **"office_city": "Vancouver",**
            "office_country": "CAN",
            "office_location_code": "V6Z2G3xx",
            "position_title": "Business Application Analyst",
            "supervisor_first_name": "Stella",
            "supervisor_last_name": "Archer"
        }

note: mask some data value with xx, just to keep original data hidden for privacy concern if any.

**Comment by HasanSifat** [comment]
@bhumin-fw  yes lets go with "city" and for the supervisor field- lets prepopulate them so yeah you can go ahead and hook up the api endpoints

**Comment by bhumin-fw** [comment]
@HasanSifat updated the below details in the SL review 2.0 
1. **ministry** (populated with **Organization** value received from ODS)
2. ADM/supervisor name and email are populated with **supervisor_name/email** values from the ODS data.
3. work city now populates the value of "**city**" key from ODS data instead of **office_city**

**Comment by HasanSifat** [comment]
tested successfully in DEV. Ready to be transferred to TEST/PROD

**Comment by MeghanStothers** [comment]
Approved - but not working for Stella (tested by Meghan and Hasan with success).

**Comment by HasanSifat** [comment]
tested successfully in TEST
